ops/pebble.py

```diff
@@ -1899,6 +1899,21 @@ def get_services(self, names: Optional[Iterable[str]] = None) -> List[ServiceInf
         resp = self._request('GET', '/v1/services', query)
         return [ServiceInfo.from_dict(info) for info in resp['result']]
 
+    if typing.TYPE_CHECKING:
+        @typing.overload
+        def pull(self,  # noqa
+                 path: str,
+                 *,
+                 encoding: None) -> BinaryIO:
+            ...
+
+        @typing.overload
+        def pull(self,  # noqa
+                 path: str,
+                 *,
+                 encoding: str = 'utf-8') -> TextIO:
+            ...
+
```

Comment:
Should we copy this to the model.Container versions too?

Reply:
Yes, good call :smile: